### PR TITLE
dev refactor geoip_enricher to use base class

### PR DESCRIPTION
**Pull request by ekneg54**
part of #41 

Please expecialy focus on the mock of `geoip2.database.Reader`

### CODE REVIEW

**Comment by ekneg54** [comment]
rebased from main

**Comment by ekneg54** [comment]
rebased from new main

File: logprep/processor/base/processor.py [hunk 1]
```diff
@@ -122,7 +122,11 @@ class RuleBasedProcessor(BaseProcessor):
     def __init__(self, name: str, tree_config: str, logger: Logger):
         super().__init__(name, logger)
         self._rules = []
+        self._specific_rules = []
```
Comment (urdit52) [inline logprep/processor/base/processor.py:125]: where are these used?

Comment (urdit52) [reply logprep/processor/base/processor.py:125]: (commenting these out does not change anything about the test results)

File: logprep/processor/geoip_enricher/processor.py [hunk 2]
```diff
@@ -27,52 +30,68 @@ class GeoIPEnricherError(BaseException):
...
         if self._logger.isEnabledFor(DEBUG):
-            self._logger.debug(f'{self.describe()} loaded {self._tree.rule_counter} rules '
-                               f'({current_process().name})')
+            self._logger.debug(
+                f"{self.describe()} loaded {self._specific_tree.rule_counter} "
+                f"specific rules ({current_process().name})"
+            )
+            self._logger.debug(
+                f"{self.describe()} loaded {self._generic_tree.rule_counter} generic rules "
+                f"({current_process().name})"
```
Comment (urdit52) [inline logprep/processor/geoip_enricher/processor.py:88]:
```suggestion
                f"generic rules ({current_process().name})"
```

File: logprep/processor/list_comparison/processor.py [hunk 3]
```diff
@@ -97,8 +97,7 @@ def add_rules_from_directory(
                 f"({current_process().name})"
             )
         self.ps.setup_rules(
-            [None] * self._generic_tree.rule_counter
-            + [None] * self._specific_tree.rule_counter
+            [None] * self._generic_tree.rule_counter + [None] * self._specific_tree.rule_counter
```
Comment (urdit52) [inline logprep/processor/list_comparison/processor.py:100]: why are we taking back the format change?

File: logprep/processor/list_comparison/processor.py [hunk 4]
```diff
@@ -116,9 +115,7 @@ def _load_rules_from_file(self, list_comparison_path: str):
         try:
             return ListComparisonRule.create_rules_from_file(list_comparison_path)
         except InvalidRuleDefinitionError as error:
-            raise InvalidRuleFileError(
-                self._name, list_comparison_path, str(error)
-            ) from error
+            raise InvalidRuleFileError(self._name, list_comparison_path, str(error)) from error
```
Comment (urdit52) [inline logprep/processor/list_comparison/processor.py:118]: see above

**Review by urdit52 — CHANGES_REQUESTED** [review]
There are a few minor details to be checked (unused lists, format changes of already formatted code). Could you please investigate?

Otherwise: lgtm, like the mock solution provided here.

File: logprep/processor/base/processor.py (hunk 1 above)
Comment (ekneg54) [reply logprep/processor/base/processor.py:125]: you have to define it in all childs. so it is better to define it here.

File: logprep/processor/list_comparison/processor.py (hunk 3 above)
Comment (ekneg54) [reply logprep/processor/list_comparison/processor.py:100]: black does his thing. And 100 chars are 100 chars. :-)
your black configuration does not respect the pyproject.toml in project root. you have to fix it.

File: logprep/processor/list_comparison/processor.py (hunk 4 above)
Comment (ekneg54) [reply logprep/processor/list_comparison/processor.py:118]: same as above. fix your black configuration

File: logprep/processor/base/processor.py (hunk 1 above)
Comment (urdit52) [reply logprep/processor/base/processor.py:125]: ok.

**Review by urdit52 — APPROVED** [review]

**Comment by ekneg54** [comment]
rebased from main

**Review by ppcad — APPROVED** [review]

**Pull request merged**